ffi/blitbuffer.lua

```diff
@@ -1826,6 +1826,109 @@ function BB_mt.__index:writePNGFromBGR(filename)
     C.free(cdata)
 end
 
+function BB_mt.__index:writeBMP(filename)
+    local function write_uint32(target_ptr, target_pos, data)
+        target_ptr[target_pos] = band(data, 0xFF)
+        target_ptr[target_pos + 1] = band(rshift(data, 8), 0xFF)
+        target_ptr[target_pos + 2] = band(rshift(data, 16), 0xFF)
+        target_ptr[target_pos + 3] = band(rshift(data, 24), 0xFF)
+        return target_pos + 4
+    end
+
+    local output_channels = 3
+    local w, h = self:getWidth(), self:getHeight()
+    local stride = w * output_channels
+
+    -- we do this before all buffer alocations
+    local of, err = io.open(filename, "wb")
+    if err ~= nil then
+        return err
+    end
+
+    local bbdump
+    local source_ptr
+    if self:getType() == TYPE_BBRGB24 then
+        source_ptr = ffi.cast("unsigned char *", self.data)
+    else
+        bbdump = BB.new(w, h, TYPE_BBRGB24, nil)
+        bbdump:blitFrom(self)
+        source_ptr = ffi.cast("unsigned char *", bbdump.data)
+    end
+
+    local filesize = stride * h + 54
+    local padding = 0
+    -- update filesize, if stride is not a multiple of 4
+    if band(stride, 3) ~= 0 then
+        padding = 4 - band(stride, 3)
+        filesize = filesize + padding * h
+    end
+
+    local target_buff = C.calloc(filesize, 1) -- initiate array with zero
+    local target_ptr = ffi.cast("unsigned char*", target_buff)
```

Comment:
nit: You can use `uint8pt` instead of `"unsigned char*"`. (It's from a cached `ffi.typeof` call).